Move govet verify into rest of verify*-.sh scripts

cblecker opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#write-release-notes-if-needed
-->

**What this PR does / why we need it**:
Instead of having two govet scripts, consolidate them to into one and have both the Makefile and verify.sh scripts target the same script. This also will allow proper syntax highlighting and timing when the vet script is run as part of `make verify`.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/kubernetes/test-infra/issues/2725

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
NONE
```

/cc @fejta @rmmh

cblecker commented:
@k8s-bot gce etcd3 e2e test this

rmmh commented on hack/verify-govet.sh:
```diff
@@ -14,24 +14,36 @@
+
+# This is required before we run govet for the results to be correct.
+# See https://github.com/golang/go/issues/16086 for details.
+make generated_files
```
this feels fishy

cblecker replied:
Github doesn't show it, but I literally just moved `hack/make-rules/vet.sh` to `hack/verify-govet.sh` and changed the BASH_SOURCE line to have one less `/..`

rmmh commented on Makefile:
```diff
@@ -341,7 +340,7 @@ vet:
 	@echo "$$VET_HELP_INFO"
 else
 vet:
```
let's make this depend on verify_generated_files too

rmmh commented:
For context, #25978 by @thockin moved this out.

I'm not sure that the recursive make is the best/safest solution here.

cblecker commented:
Hmm okay. The picture is becoming a bit clearer. Let me see what I can do.

fejta commented:
/approve
/uncc
/unassign @jbeda @ixdy 
/assign @rmmh

cblecker commented:
/cc @ixdy 
In tracing back some of the history here, the reason generated_files needs to be run first looks like a go issue with how vet works (something @ixdy opened an issue with the golang team about). I believe the way the way I've changed the scripts around should work in a bit of a safer way, and get us to the goals of:
- run steps like this using make so that eventually we can move away from committing generated files
- run go vet using the verify.sh script in CI, so that FAILED/SUCCESS and timing messages are put into the build log.

cblecker commented:
@k8s-bot pull-kubernetes-federation-e2e-gce test this
@k8s-bot kops aws e2e test this

cblecker commented:
@k8s-bot pull-kubernetes-federation-e2e-gce test this

cblecker commented:
From my testing PR (https://github.com/kubernetes/kubernetes/pull/46155):
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/46155/pull-kubernetes-verify/30855/
```
I0519 18:05:12.943] Verifying hack/make-rules/../../hack/verify-govet.sh
W0519 18:05:38.326] cmd/kube-proxy/app/conntrack.go:124: wrong number of args for format in Errorf call: 1 needed but 2 args
W0519 18:05:38.328] exit status 1
I0519 18:06:12.497] Makefile:343: recipe for target 'vet' failed
I0519 18:06:12.499] FAILED   hack/make-rules/../../hack/verify-govet.sh	60s
I0519 18:06:12.499] Verifying hack/make-rules/../../hack/verify-import-boss.sh
W0519 18:06:12.600] make[1]: *** [vet] Error 1
[..snip..]
I0519 18:19:52.709] Makefile:123: recipe for target 'verify' failed
W0519 18:19:52.809] make: *** [verify] Error 1
```

cblecker commented:
@k8s-bot pull-kubernetes-federation-e2e-gce test this

cblecker commented:
@rmmh @ixdy Is there a chance of this getting reviewed before code freeze (EOD 6/1)?

rmmh commented:
/lgtm

cblecker commented:
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

cblecker commented:
/assign @ixdy 
for approval :)

ixdy commented on hack/make-rules/vet.sh:
```diff
@@ -25,8 +25,14 @@ cd "${KUBE_ROOT}"
-go install ./cmd/...
+# If called directly, exit.
+CALLED_FROM_MAIN_MAKEFILE=${CALLED_FROM_MAIN_MAKEFILE:-""}
+if [ "${CALLED_FROM_MAIN_MAKEFILE}" == "" ]; then
```
I might've just done `if [[ -z "${CALLED_FROM_MAIN_MAKEFILE:-}" ]]; then` since it's shorter, but this is fine too.

ixdy commented on hack/make-rules/vet.sh:
```diff
@@ -25,8 +25,14 @@ cd "${KUBE_ROOT}"
 
 # This is required before we run govet for the results to be correct.
```
this comment probably needs to be updated - "This" no longer refers to anything.

cblecker replied:
Yup, makes sense.

ixdy commented on hack/make-rules/vet.sh:
```diff
@@ -25,8 +25,14 @@ cd "${KUBE_ROOT}"
 # This is required before we run govet for the results to be correct.
 # See https://github.com/golang/go/issues/16086 for details.
-make generated_files
-go install ./cmd/...
```
ISTR the `go install` bit was important too, though I'm not sure if it still is. this is my main concern with this PR.

cblecker replied:
Reviewing that github issue, it looks like it still might be important, until at least Go 1.9

cblecker commented:
@ixdy @rmmh Updated per comments. PTAL

cblecker commented:
@k8s-bot pull-kubernetes-cross test this
@k8s-bot pull-kubernetes-federation-e2e-gce test this

cblecker commented:
@k8s-bot pull-kubernetes-federation-e2e-gce test this

fejta commented:
@k8s-bot pull-kubernetes-federation-e2e-gce test this
ref: #46827

ixdy commented:
/lgtm

cblecker commented:
@ixdy rebased

cblecker commented:
/retest

ixdy commented on hack/make-rules/vet.sh:
```diff
@@ -23,9 +23,16 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 cd "${KUBE_ROOT}"
 
+# If called directly, exit.
+if [ "${CALLED_FROM_MAIN_MAKEFILE:-""}" == "" ]; then
```
nit: I think you might need to use `[[ ]]` for `==` to work? we generally prefer `[[ ]]` anyway.

cblecker replied:
I verified it worked without, but cleaned it up anyways to stick with convention.

cblecker committed:
Run hack/verify-govet.sh as part of verify make target
This commit ensures that:
- go vet will be run as part of the make verify target
- the vet make-rule script won't be run directly, as generated_files won't be run in that case
- that go vet errors show up in the build log with a start time, finish time, and SUCCESS/FAILED message as part of the verify make rule script

cblecker commented:
/test pull-kubernetes-e2e-gce-etcd3

ixdy commented:
/lgtm
